examples/dask/binary-classification.py

```diff
@@ -0,0 +1,30 @@
+import dask.array as da
+from distributed import Client, LocalCluster
+from sklearn.datasets import make_blobs
+
+import lightgbm as lgb
+
+if __name__ == "__main__":
+
+    print("loading data")
+
+    X, y = make_blobs(n_samples=1000, n_features=50, centers=2)
+
+    print("initializing a Dask cluster")
+
+    cluster = LocalCluster()
+    client = Client(cluster)
+
+    print("created a Dask LocalCluster")
+
+    print("distributing training data on the Dask cluster")
+
+    dX = da.from_array(X, chunks=(100, 50))
+    dy = da.from_array(y, chunks=(100,))
+
+    print("beginning training")
+
+    dask_model = lgb.DaskLGBMClassifier()
+    dask_model.fit(dX, dy)
```

Comment:
```suggestion
    dask_model.fit(dX, dy)
    assert dask_model.fitted_
```

Reply:
I don't believe user-facing examples intended to show how to use this code should have asserts like this in it. Why do you think we need them?

By the way none of the examples in https://github.com/microsoft/LightGBM/tree/master/examples/python-guide have asserts like this

Reply:
I agree that it is uncommon to see any `assert`s in examples. But as we are running examples at our CI, these `assert`s make examples code more reliable. Also, I think it is good from the users side view to have some asserts because in case of any problems (installation especially) it is easier to check what is going wrong with intermediate asserts. But not just something like "the whole example isn't working".

> By the way none of the examples in https://github.com/microsoft/LightGBM/tree/master/examples/python-guide have asserts like this

Those examples are much more complex and a lot of issues can be spotted via "`predict()` will fail without correctly executed `fit()`". In Dask almost all examples are just about "fit model" and there are no any other actions that can check that `fit` was really performed. Also, `fitted_` was introduced quite recently (at least, much more later than those examples were created).

Reply:
alright, I won't hold up the PR over this. I can add them.